### DISCUSSION TITLE
Wind ramp rates

### DISCUSSION THREAD

**Discussion by bendnorman** [catalyst-cooperative/pudl#3644] [Help Me!] [answered]
Does PUDL have any data on wind generator ramp rates?

**Comment by zaneselvans** [comment, marked as answer]
Can you say more about what this would mean? Like the observed rates at which wind power comes and goes? Or would this be an attribute of the generator?  At the individual generator level? Or plant (wind farm)? Or within some geographic region?

I would look at the wind generator table attributes from EIA-860 if it's a function of the turbine.  For observed ramp rates the best we'll have is regional and hourly, either from the GridPath RA Toolkit renewable generation profiles, or the hourly EIA-930 data and the ISO level if they report wind broken out as a separate energy source.

**Reply by zaneselvans** [reply]
The as-of-yet unintegrated GridPath RA Toolkit data also has unaggregated hourly plant-level wind capacity factors for the WECC region, if you wanted to look at pulling that in or using it as is.  It's large but not structurally complicated.

**Reply by TrentonBush** [reply]
Any hourly plant-level data are almost certainly derived from a weather model. Real operational data is proprietary ☹️

**Reply by zaneselvans** [reply]
Yeah, it's based on the NREL WINDS toolkit which gives the production curve, and then tuned to reproduce the monthly aggregate plant level generation that's reported in the EIA-923.

**Reply by bendnorman** [reply]
Thanks for the info! Is the GridPath RA Toolkit with ramp rates integrated into PUDL yet?

**Reply by zaneselvans** [reply]
Hourly available capacity factors aggregated to the BA-ish level are available in the `out_gridpathratoolkit__hourly_available_capacity_factor` table. Modeled production curves for the individual plants (which are aggregated up to create that table) have not yet been integrated.

**Comment by TrentonBush** [comment]
For an individual plant, I don't think so, no. That would require operational data with hourly resolution, if not finer. We have hourly operational data for fossil plants thanks to EPA CEMS, but non-emitting generators like wind are not included there.

Back when I was a wind analyst, I remember somebody presenting research that compared real wind ramp rates vs ramp rates estimated from hourly MERRA reanalysis meteorological data. If I remember correctly, the derived values underestimated the real ramp rates (likely due to the coarse spatial/temporal resolution of MERRA), but could be useful as a lower bound. That might not be a bad starting place.
